tic-tac-toe.js

```diff
@@ -1,188 +1,172 @@
-let players = ["player1", "player2"];
-let states = ["open", "player1won", "player2won", "draw"];
+const Marks = {
+  P1: "X",
+  P2: "O",
+  Empty: ".",
+};
+
+const players = {
+  P1: "player1",
+  P2: "player2",
+};
+
+const states = {
+  open: "open",
+  P1: "player1won",
+  P2: "player2won",
+  dran: "draw",
+};
+
+/*const players = ["player1", "player2"];
+const states = ["open", "player1won", "player2won", "draw"];
+const squareStates = ["not set", "player 1", "player 2"];*/
 
-class TicTacToe {
-    constructor(){
+export class TicTacToe {
+  // Saninn's comment: Usually the functions should be defined AFTER the constructor... but ok...
+  resetGame() {
     this.field = [
-        [".", ".", "."],
-        [".", ".", "."],
-        [".", ".", "."]
-        ];
-    this.squareState = ["not set", "player 1", "player 2"];
-    /* I HAVEN'T USED THIS*/
-    this.currentPlayer = players[0];
-    this.currentState = states[0];
-    this.movesArray = [];
-    };
-    
-    resetGame() {
-        this.field = [
-            [".", ".", "."],
-            [".", ".", "."],
-            [".", ".", "."]
-            ];
-        this.currentPlayer = players[0];
-        this.currentState = states[0];
+      [Marks.Empty, Marks.Empty, Marks.Empty],
+      [Marks.Empty, Marks.Empty, Marks.Empty],
+      [Marks.Empty, Marks.Empty, Marks.Empty],
+    ];
+    this.currentPlayer = players.P1;
+    this.currentState = states.open;
+  }
+
+  constructor() {
+    this.resetGame();
+  }
+
+  _playerHasWon(userMark) {
+    // returns boolean
+    if (
+      (this.field[0][0] === userMark) &
+        (this.field[0][1] === userMark) &
+        (this.field[0][2] === userMark) ||
+      (this.field[1][0] === userMark) &
+        (this.field[1][1] === userMark) &
+        (this.field[1][2] === userMark) ||
+      (this.field[2][0] === userMark) &
+        (this.field[2][1] === userMark) &
+        (this.field[2][2] === userMark) ||
+      (this.field[0][0] === userMark) &
+        (this.field[1][0] === userMark) &
+        (this.field[2][0] === userMark) ||
+      (this.field[0][1] === userMark) &
+        (this.field[1][1] === userMark) &
+        (this.field[2][1] === userMark) ||
+      (this.field[0][2] === userMark) &
+        (this.field[1][2] === userMark) &
+        (this.field[2][2] === userMark) ||
+      (this.field[0][0] === userMark) &
+        (this.field[1][1] === userMark) &
+        (this.field[2][2] === userMark) ||
+      (this.field[2][0] === userMark) &
+        (this.field[1][1] === userMark) &
+        (this.field[0][2] === userMark)
```

Comment:
I have the feeling we can still do this more clean with some kind of clever math play. But it Works! So ignore this 😄

Reply:
Alright.